engine/src/main/java/io/zeebe/engine/processor/workflow/deployment/model/validation/ZeebeExpressionValidator.java

```diff
@@ -142,10 +133,16 @@ public Builder(final Class<T> elementType) {
       return this;
     }
 
-    public Builder<T> hasValidExpressionIfPresent(final Function<T, String> expressionSupplier) {
+    public Builder<T> hasValidNonStaticExpressionIfPresent(
```

Comment:
Optional comment:

The naming is a bit inconvenient, in my mind.
FIrst the "NonStatic" It would be nice to have a positive variant, like "DynamicExpression" or something like that. But I cannot really come up with a good name. And "Dynamic" does not really cut it.

The second issue is the naming of `hasValidNonStatic` vs. `hasValidNonStaticIfPresent`. Here you could use `hasValidMandatoryNonStaticExpression` vs. `hasValidOptionalNonStaticExpression` which captures the semantics better, in my mind. It's more of a preference, though